community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/AbstractKeyValueStore.java

```diff
@@ -317,7 +322,13 @@ private long rotate( boolean force ) throws IOException
                     // won't close the state as it was before rotation began, which we're reverting to right here.
                     try ( LockWrapper ignored = writeLock( updateLock ) )
                     {
-                        state = rotation.markAsFailed();
+                        // Only mark as failed if we're still running, because if this store has been stopped
+                        // this rotation task will be the only option to close the preState and marking
+                        // this rotation as failed will prevent preState to be closed.
+                        if ( !stopped )
+                        {
+                            state = rotation.markAsFailed();
+                        }
```

Comment:
I'm not quite seeing how this helps, and the comment isn't quite helping me either.

It seems to me as if the store would already be in a stopped state if the `stopped` flag is `true`, since we both assigned the `stopped` flag and invoked `stop()` on the state under the same lock. Here we are also under the `updateLock`, so we should atomically see that view of the world. In that case, we should already be in the `Stopped` state, in which case rotation is not allowed, so we would not reach here.

The changes in this commit are thus pointless, and only add confusion.